### PR TITLE
soroban-rpc: Add contract events to simulateTransaction's response

### PR DESCRIPTION
**Pull request by 2opremio**
### What

Add contract events to simulateTransaction's response. This change modifies libpreflight to return contract events and the `simulateTransaction` JSONRPC method to include contract events in its response.

### Why

Closes https://github.com/stellar/soroban-tools/issues/475

### Known limitations

N/A

TODO:
* [ ] Update soroban-rpc documentation at https://github.com/stellar/soroban-docs

### CODE REVIEW

File: cmd/soroban-rpc/internal/preflight/preflight.go [hunk 1]
```diff
@@ -97,12 +97,31 @@ type PreflightParameters struct {
...
 }
 
+// GoNullTerminatedStringSlice transforms a C NULL-terminated char** array to a Go string slice
+func GoNullTerminatedStringSlice(str **C.char) []string {
+	var result []string
+	if str != nil {
+		// CGo doesn't have an easy way to do pointer arithmetic so,
+		// we are better off transforming the memory buffer into a large slice
+		// and finding the NULL termination after that
+		for _, a := range unsafe.Slice(str, 1<<20) {
```
[the remark this answers is not in the document]
Comment (paulbellamy) [reply cmd/soroban-rpc/internal/preflight/preflight.go:114]: Basically, i worry that we accidentally re-use this method for a value > 1mb in the future, and it fails mysteriously.

File: cmd/soroban-rpc/lib/preflight/src/lib.rs [hunk 2]
```diff
@@ -185,7 +188,7 @@ fn preflight_host_function_or_maybe_panic(
     let auth_payloads = host.get_recorded_auth_payloads()?;
 
     // Recover, convert and return the storage footprint and other values to C.
-    let (storage, budget, _) = host.try_finish().unwrap();
+    let (storage, budget, events) = host.try_finish().unwrap();
```
Comment (paulbellamy) [inline cmd/soroban-rpc/lib/preflight/src/lib.rs:191]: Side question: Does this include diagnostic events? or are those emitted somewhere else?

**Review by paulbellamy — APPROVED** [review]

**Comment by 2opremio** [comment]
Closing in favor of #536

**Comment by 2opremio** [comment]
My bad, I should had closed https://github.com/stellar/soroban-tools/pull/535

File: cmd/soroban-rpc/lib/preflight/src/lib.rs (hunk 2 above)
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/lib.rs:192]: Good question. Now that you bring it up, I am not sure. @sisuresh @jayz22 ?

Comment (sisuresh) [reply cmd/soroban-rpc/lib/preflight/src/lib.rs:192]: Yeah these include diagnostic events.

File: cmd/soroban-rpc/internal/preflight/preflight.go [hunk 3]
```diff
@@ -96,11 +96,30 @@ type PreflightParameters struct {
 }
 
 type Preflight struct {
-	Auth            []string // ContractAuths XDR in base64
-	Footprint       string   // LedgerFootprint XDR in base64
-	Result          string   // SCVal XDR in base64
-	CPUInstructions uint64
-	MemoryBytes     uint64
+	Auth             []string // ContractAuths XDR in base64
+	DiagnosticEvents []string // DiagnosticEvents XDR in base64
```
Comment (2opremio) [inline cmd/soroban-rpc/internal/preflight/preflight.go:100]: Maybe we should just call this `Events`?

```suggestion
	Events []string // DiagnosticEvents XDR in base64
```

File: cmd/soroban-rpc/internal/preflight/preflight.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (2opremio) [reply cmd/soroban-rpc/internal/preflight/preflight.go:114]: Sure, I can change it.

Note that it's not really 1MB of memory but 2^20 entries in the slice.

File: cmd/soroban-rpc/lib/preflight/src/lib.rs (hunk 2 above)
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/lib.rs:192]: great!

File: cmd/soroban-rpc/internal/preflight/preflight.go (hunk 3 above)
Comment (2opremio) [reply cmd/soroban-rpc/internal/preflight/preflight.go]: Done

File: cmd/soroban-rpc/internal/preflight/preflight.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (2opremio) [reply cmd/soroban-rpc/internal/preflight/preflight.go]: When using `math.MaxInt` the code panics with `unsafe.Slice: len out of range` (And even with `math.MaxInt`/2).

I think 1 million strings should be more than enough in any practical case.

**Pull request merged**